clients/privacy-center/cypress/e2e/banner-overlay-dismissal.cy.ts

```diff
@@ -0,0 +1,358 @@
+        cy.get("#fides-banner").should("be.visible");
+        cy.get("h1").click(); // click outside
+        cy.get("#fides-banner").should("not.be.visible");
+
+        cy.get("@FidesUpdated")
+          .should("have.been.calledOnce")
+          .its("lastCall.args.0.detail.extraDetails.consentMethod")
+          .then((consentMethod) => {
+            expect(consentMethod).to.eql(ConsentMethod.dismiss);
+          });
+      });
+
+      it("Should dismiss the banner by hitting ESC", () => {
+        cy.get("#fides-banner").should("be.visible");
+        cy.get("body").type("{esc}");
+        cy.get("#fides-banner").should("not.be.visible");
+
+        cy.get("@FidesUpdated")
+          .should("have.been.calledOnce")
+          .its("lastCall.args.0.detail.extraDetails.consentMethod")
+          .then((consentMethod) => {
+            expect(consentMethod).to.eql(ConsentMethod.dismiss);
+          });
+      });
+    });
+
+    describe("Overlay dismissal", () => {
+      it("Should dismiss the banner by clicking the x", () => {
+        cy.get("#fides-banner").should("be.visible");
+        cy.getByTestId("Manage preferences-btn").click();
+        cy.get(".fides-modal-content").should("be.visible");
+        cy.get(".fides-modal-content").within(() => {
+          cy.get(".fides-close-button").click();
+        });
+        cy.get(".fides-modal-content").should("not.be.visible");
+
+        cy.get("@FidesUpdated")
+          .should("have.been.calledOnce")
+          .its("lastCall.args.0.detail.extraDetails.consentMethod")
+          .then((consentMethod) => {
+            expect(consentMethod).to.eql(ConsentMethod.dismiss);
+          });
+      });
+
+      it("Should dismiss the banner by clicking outside", () => {
+        cy.get("#fides-banner").should("be.visible");
+        cy.getByTestId("Manage preferences-btn").click();
+        cy.get(".fides-modal-content").should("be.visible");
+        cy.get(".fides-modal-overlay").click({ force: true });
+        cy.get(".fides-modal-content").should("not.be.visible");
+
+        cy.get("@FidesUpdated")
+          .should("have.been.calledOnce")
+          .its("lastCall.args.0.detail.extraDetails.consentMethod")
+          .then((consentMethod) => {
+            expect(consentMethod).to.eql(ConsentMethod.dismiss);
+          });
+      });
+
+      it("Should dismiss the banner by hitting ESC", () => {
+        cy.get("#fides-banner").should("be.visible");
+        cy.getByTestId("Manage preferences-btn").click();
+        cy.get(".fides-modal-content").should("be.visible");
+        cy.get("body").type("{esc}");
+        cy.get(".fides-modal-content").should("not.be.visible");
+
+        cy.get("@FidesUpdated")
+          .should("have.been.calledOnce")
+          .its("lastCall.args.0.detail.extraDetails.consentMethod")
+          .then((consentMethod) => {
+            expect(consentMethod).to.eql(ConsentMethod.dismiss);
+          });
```

Comment:
might be worth putting this in a function, along with an assertion that the PATCH was called, since we check for it every time. maybe a function like `assertDismissCalled`?